### PR TITLE
Catalogue of my things - Ruby Capstone Project

### PR DESCRIPTION
**Pull request by maov19**
### Logic
1. Created all UML classes required and its relationships:
- _Label, Genre, Author_
- _Book, Music Album, Game_
2. Implemented all methods required.
3. Added unit tests for all implemented methods

### User interface
1. Created a main.rb file as console app entry-point, displaying required options.
2. Preserved all data by saving collections in .json files.

### Database
1. Created a schema.sql file with tables that will be analogical to the structure of the classes in your app. 

### Presentation
**[Link to project's intro video](https://drive.google.com/file/d/1REg2-XZwsNvMMal9g9Q1W7-xdD_Gk3XP/view?usp=sharing)**

Thanks! 👋

### CODE REVIEW

File: music.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  attr_accessor :name, :genres, :publish_date, :on_spotify
+  attr_writer :genre
+
+  def initialize(name, publish_date, on_spotify: false, genres: [])
+    super(publish_date)
+    @name = name
+    @publish_date = Date.parse(publish_date)
+    @on_spotify = on_spotify
+    @genres = genres
+  end
```
Comment (anagudelogu) [inline music.rb:14]: - [ ] Remember that the relationship between `Item` and `Genre` is one-to-many, for instance, this `genres` array makes no sense here, as a Music Album belongs to only one Genre. Please consider fixing this! 🤓 🔧

File: spec/genre_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  it 'adds itself as the item\'s genre' do
+    @genre.add_item(@music_album)
+    expect(@music_album.genre).to eq(@genre)
+  end
+
+  it 'adds the item to its items collection' do
+    @genre.add_item(@music_album)
+    expect(@genre.items).to include(@music_album)
+  end
```
Comment (anagudelogu) [inline spec/genre_spec.rb:34]: - [ ] Please make sure to follow the DRY principle, here you have duplicated tests that check if the item gets added to the @items collection. Please consider cleaning this! 🤓 🔧

File: spec/label_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  it 'adds itself as the item\'s label' do
+    @label.add_item(@book)
+    expect(@book.label).to eq(@label)
+  end
+
+  it 'adds the item to its items collection' do
+    @label.add_item(@book)
+    expect(@label.items).to include(@book)
+  end
```
Comment (anagudelogu) [inline spec/label_spec.rb:33]: - [ ] Please make sure to follow the DRY principle, here you have duplicated tests that check if the item gets added to the @items collection. Please consider cleaning this! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi Team! @maov19 @Koshao-sys @raihan2bd, 
I'm Andres.

While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

### VIDEO PRESENTATION 🎬  

You did a great job working on your video presentation! 💯 However, there's still room to improve:

- [ ] According to the requirements, the maximum length of the video should be 5 min. Please consider staying under that time! 🤓 🔧 

- [ ] I suggest that you plan ahead the outline of what you're going to talk about, take in mind that being capable to speak about your projects is one of the best skills to attract future recruiters. Since you're a group of three, here's my suggested outline:

1.  Brief presentation of the team. 👔 (Student A)
2.  Brief explanation of the purpose of the project, (what was the main motivation: to solve a problem, practice, showcase a skill, etc). 📜 (Student A)
3.  Go through a live demo of all the features of your project, but don't spend too much time explaining implementation details (non-tech people will thank you). 🤖 (Student B)
4.  Finish with a piece of code that you think is interesting or that makes you proud! 🥇 (Student C)

Remember that you're doing the presentation as a team, so rather than presenting what each one of you did in order to build the project, present the finished product, and what made you proud as a team! 🤓 🔧 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/label_spec.rb (hunk 3 above)
Comment (maov19) [reply spec/label_spec.rb]: Fixed!

File: spec/book_tests_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    end
+
+    context 'when cover state is "bad"' do
+      it 'returns true' do
+        book = Book.new('Test Book', Date.today, 'Test Publisher', 'bad')
+        expect(book.can_be_archived?).to eq(true)
+      end
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (maov19) [reply spec/book_tests_spec.rb]: Fixed!

File: spec/add_book_spec.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+require_relative '../book_app'
+require_relative '../book'
+require_relative '../label'
+require 'date'
+require 'json'
+
+RSpec.describe BookActions do
```
[the remark this answers is not in the document]
Comment (maov19) [reply spec/add_book_spec.rb:7]: fixed!

File: genre.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  def initialize(name)
+    @id = Random.new.rand(1..1000)
+    @name = name
+    @items = []
+  end
+
+  def add_item(item)
+    @items.push(item)
+    item.genre = self
+    item.add_genre(self)
```
[the remark this answers is not in the document]
Comment (raihan2bd) [reply genre.rb]: @anagudelogu Thanks a lot for your suggestion 👍

**Comment by maov19** [comment]
Hi @anagudelogu!

Thanks a a lot for the feedback! We already fixed all the mentioned issues.
1. **Video**: we uploaded a new version.
> Thanks a lot for your guidelines, they were very useful
2. **Components:** defs on the following classes were fixed and are 100% functional.
> music.rb 
>genre.rb 
>item.rb
> game.rb 
3. **Testing:** the following tests files were fixed:
> add_book_spec.rb
> book_tests_spec.rb
> author_spec.rb
> book_spec.rb
> genre_spec.rb
> label_spec.rb

Please, let us know if there is something else we should work on 👍

**Review by codecaiine — APPROVED** [review]
Hi @maov19 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Great use of Git flow. ✔️ 
- Your project is documented professionally. ✔️ 
- All test are passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**